examples/docker-rootful.yaml

```
@@ -44,16 +44,64 @@ provision:
     #!/bin/bash
     set -eux -o pipefail
     command -v docker >/dev/null 2>&1 && exit 0
-    if [ ! -e /etc/systemd/system/docker.socket.d/override.conf ]; then
-      mkdir -p /etc/systemd/system/docker.socket.d
+    readonly override_conf=/etc/systemd/system/docker.socket.d/override.conf
+    if [ ! -e "$override_conf" ]; then
+      mkdir -p $(dirname "$override_conf")
       # Alternatively we could just add the user to the "docker" group, but that requires restarting the user session
-      cat <<-EOF >/etc/systemd/system/docker.socket.d/override.conf
-      [Socket]
-      SocketUser={{.User}}
+      cat <<EOF >"$override_conf"
+    [Socket]
+    SocketUser={{.User}}
     EOF
     fi
     export DEBIAN_FRONTEND=noninteractive
     curl -fsSL https://get.docker.com | sh
+- mode: user  # configure docker under non-root user
+  script: |
+    #!/bin/bash
+    set -eux -o pipefail
+    command -v jq &>/dev/null || sudo apt-get install -y jq
+    readonly rootless_installed=$(systemctl --user list-unit-files docker.service &>/dev/null && echo true || echo false)
+
+    if [ "{{.Param.Rootful}}" = "true" ]; then
+      readonly config_dir="/etc/docker"
+      readonly systemctl="sudo systemctl"
+      readonly tee="sudo tee"
+
+      [ "$rootless_installed" != "true" ] || systemctl --user disable --now docker
+      docker context use default
+
+    else
+      readonly config_dir="$HOME/.config/docker"
+      readonly systemctl="systemctl --user"
+      readonly tee="tee"
+
+      sudo systemctl disable --now docker
+      if [ "$rootless_installed" != "true" ]; then
+        sudo apt-get install -y dbus-user-session fuse3 uidmap
+        $systemctl start dbus
+        [ ! -S /var/run/docker.sock ] || sudo rm /var/run/docker.sock
+        dockerd-rootless-setuptool.sh install
+      fi
+      docker context use rootless
+    fi
+    $systemctl enable --now docker
+
+    readonly config="$config_dir/daemon.json"
+    needs_restart=
```

You use `true` and `false` for the `param.*` booleans, so I would suggest to do the same for other variables.

```suggestion
    needs_restart=false
```